 design/README.MD

```diff
@@ -0,0 +1,456 @@
+## Overview
+
+Smithy is an interface definition language.
+It lets you abstract your interface
+from any given executable implementation.
+By focusing on the abstract interface
+Smithy helps you build an interface
+that can safely evolve
+maintaining both forward and backwards compatibility.
+
+Dafny is a programming language with formal verification features.
+Dafny code is translated to higher-level languages
+to provide interoperability for your verified code
+like Java or C#.
+These native representations are then executable.
+This means that a single Dafny program
+can be translated into multiple interoperable programs.
+
+Dafny itself has a `string` type
+that you might assume can map
+to the obvious string type in each translated language.
+
+However a native `string` may be UTF8, UTF16/USC2, or UTF32 
+depending on the language.
+Numbers are similarly complicated.
+Some languages have different sized numbers,
+while others have either a different default or a single size.
+There are also complicated types like streams.
+All these languages have equivalent
+but not congruent types to each other.
+
+In this project Dafny and Smithy are combined
+to generate concrete interfaces in Dafny
+and supported Dafny translation targets.
+These generated interfaces all use the same translated Dafny implementation
+and expose native interfaces and types for each language.
```

Comment:
nit: IMO it might be clearer to word it like

> this allows a single implementation in Dafny to generate native interfaces in other languages

since it begins with what seems like the most important part and is structured to match the process of using the tool (write Dafny -> generate native libraries -> use libraries).